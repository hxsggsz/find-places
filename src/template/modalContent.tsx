import { Modal } from '../components/modal/modal'
import { Eye, EyeClosed, X } from 'phosphor-react'
import { StyledEye } from '.'
import { useState } from 'react'

export const ModalContent: React.FC = () => {
  const [modal, setModal] = useState(false)
  return (
    <>
      {modal && (
        <Modal.Root >
          <Modal.Container>
            <Modal.Title >
              <h1>Sobre o projetoðŸ’ª</h1>
              <Modal.Icon onClick={() => setModal(false)}>
                <X data-testid='close' size={20} />
              </Modal.Icon>
            </Modal.Title>
            <Modal.Body>
              <h1>O que Ã© esse projeto?ðŸ¤“</h1>
              <p>Encontre lugares lindos e famosos por todo o mundo para visitar! Esse site Ã© um projeto de estudo Fullstack meu, criando uma API prÃ³pia com o Apollo server/GraphQL com MySQL como banco de dados</p>
            </Modal.Body>
          </Modal.Container>
        </Modal.Root>
      )}

      <StyledEye onClick={() => setModal(!modal)}>
        {modal ? <EyeClosed data-testid='closeEye' size={20} /> : <Eye data-testid='openEye' size={20} />}
      </StyledEye>
    </>
  )
}