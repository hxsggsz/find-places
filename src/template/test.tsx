import { ModalContent } from './modalContent'
import { render, waitFor, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('<ModalContent />', () => {
  it('should render the button to open modal', () => {
    const { getByTestId } = render(<ModalContent />)

    const openEye = getByTestId('openEye')

    expect(openEye).toBeInTheDocument()
  });

  it('should open modal', async () => {
    const { getByRole, queryByText, baseElement } = render(<ModalContent />)
    const openEye = getByRole('button')

    userEvent.click(openEye)
    expect(baseElement).toMatchSnapshot();

    await waitFor(() => {
      return expect(queryByText('Sobre o projetoðŸ’ª')).toBeInTheDocument()
    })
  });
});