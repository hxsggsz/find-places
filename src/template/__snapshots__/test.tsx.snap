// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContent /> should open modal 1`] = `
.c0 {
  top: 0;
  right: 0;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 15px;
  padding: 5px;
  cursor: pointer;
  position: absolute;
  color: #333;
  border-radius: 10px;
}

.c0:hover {
  background: #111112;
  color: #fff;
}

<body>
  <div>
    <button
      class="c0"
    >
      <svg
        data-testid="openEye"
        fill="currentColor"
        height="20"
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <path
          d="M128,56C48,56,16,128,16,128s32,72,112,72,112-72,112-72S208,56,128,56Z"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <circle
          cx="128"
          cy="128"
          fill="none"
          r="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </button>
  </div>
</body>
`;
